src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemEventArgs.cs

```diff
@@ -21,14 +21,29 @@ public FileSystemEventArgs(WatcherChangeTypes changeType, string directory, stri
 
             _changeType = changeType;
             _name = name;
+            _fullPath = Combine(directory, name);
+        }
 
-            if (directory.Length == 0) // empty directory means current working dir
+        /// <summary>Combines a directory path and a relative file name into a single path.</summary>
+        /// <param name="directoryPath">The directory path.</param>
+        /// <param name="name">The file name.</param>
+        /// <returns>The combined name.</returns>
+        /// <remarks>
+        /// This is like Path.Combine, except without argument validation,
+        /// and a separator is used even if the name argument is empty.
+        /// </remarks>
+        internal static string Combine(string directoryPath, string? name)
+        {
+            bool hasSeparator = false;
+            if (directoryPath.Length > 0)
             {
-                directory = ".";
+                char c = directoryPath[directoryPath.Length - 1];
```

Comment:
Nit: what you have is fine, but it could also be:
```suggestion
                char c = directoryPath[^1];
```